agent/src/main/scala/za/co/absa/atum/agent/AtumAgent.scala

```diff
@@ -1,17 +1,39 @@
 package za.co.absa.atum.agent
 
-import za.co.absa.atum.agent.model.MeasureResult
+import za.co.absa.atum.agent.model.{AtumPartitions, MeasureResult}
 
 /**
  *  Place holder for the agent that communicate with the API.
  */
 object AtumAgent {
 
+  /**
+   *  Sends the `MeasureResult` and extra data from a given `AtumContext` to the AtumService API.
+   *  @param checkpointKey
+   *  @param atumContext
+   *  @param measureResult
+   */
+  def publish(checkpointKey: String, atumContext: AtumContext, measureResult: MeasureResult): Unit = println(
+    Seq(checkpointKey, atumContext, measureResult).mkString(" || ")
+  )
+
+  /**
+   *  Sends a single `MeasureResult` to the AtumService API. With not AtumContext involve.
+   *
+   *  @param checkpointKey
+   *  @param context
+   *  @param measureResult
+   */
   def measurePublish(checkpointKey: String, measure: MeasureResult): Unit =
     println(s"Enqueued measurement: $checkpointKey, " + (measure))
 
-  def publish(checkpointKey: String, context: AtumContext, measureResult: MeasureResult): Unit = println(
-    Seq(checkpointKey, context, measureResult).mkString(" || ")
-  )
+  /**
+   *  Provides an AtumContext given a `AtumPartitions` instance. Retrieves the data from Atum Service API.
+   *  @param atumPartitions
+   *  @return
+   */
+  def createAtumContext(atumPartitions: AtumPartitions): AtumContext = {
+    AtumContext()
```

Comment:
This would need more logic. According to my insight, the returned value should be a kind of singleton of `AtumContext`. For same `atumPartitions` the same `AtumContext` needs to be returned.
From that POV I am even considering making `AtumContext` mutable in the regard of `Measures`. What do you think guys? @seldon851, @Zejnilovic, @dk1844 , @lsulak, @jakipatryk, @yruslan